core/src/main/scala/filodb.core/metadata/Dataset.scala

```diff
@@ -42,15 +49,19 @@ object DatasetOptions {
   def fromString(s: String): DatasetOptions = {
     val config = ConfigFactory.parseString(s)
     DatasetOptions(chunkSize = config.getInt("chunkSize"),
-                   segmentSize = config.getString("segmentSize"))
+                   segmentSize = config.getString("segmentSize"),
+                   defaultPartitionKey = config.as[Option[PartitionKey]]("defaultPartitionKey"))
   }
 }
 
+/**
+ * Contains many helper functions especially pertaining to dataset partitioning.
+ */
 object Dataset {
   // If a partitioning column is not defined then this refers to a single global partition, and
   // the dataset must fit in one node.
   val DefaultPartitionColumn = ":single"
-  val DefaultPartitionKey: Types.PartitionKey = "/0"
+  val DefaultPartitionKey: PartitionKey = "/0"
 
   val DefaultOptions = DatasetOptions(chunkSize = 1000,
```

Comment:
public?  Depends on your definition of public.  Not external facing, like NodeCoordinatorActor, but public part of Dataset API, yes.